lib/active_merchant/billing/gateways/paybox_direct.rb

```diff
@@ -63,11 +63,30 @@ def initialize(options = {})
         requires!(options, :login, :password)
         super
       end
-
+      
+      def add_3dsecure(post, three_d_secure)
+        if three_d_secure.eci == '02' || three_d_secure.eci == '05'
+            post[:3DSTATUS]  = 'Y'
+            post[:3DENROLLED]  = 'Y'
+            post[:3DSIGNVAL] = 'Y'
+            post[:3DERROR] = '0'
+        else 
+            post[:3DSTATUS]  = 'N'
+            post[:3DENROLLED]  = 'N'
+            post[:3DSIGNVAL] = 'N'
+            post[:3DERROR] = '10000'
+        end
+        post[:3DECI]  = three_d_secure.eci
+        post[:3DXID] = three_d_secure.xid
+        post[:3DCAVV] = three_d_secure.cavv
+        post[:3DCAVVALGO] = '0000000001'
```

Comment:
Besides it not being listed here: https://github.com/activemerchant/active_merchant/wiki/Standardized-3DS-Fields

You could do `options[:three_d_secure][:cavv_algorithm]` and you should have the value you need, if present. (you might want to do presence checks in case the API does not accept null/blank values)